master/internal/api_generic_tasks.go

```diff
@@ -157,9 +168,45 @@ func (a *apiServer) CreateGenericTask(
 		return nil, err
 	}
 
+	var parentConfig []byte
+	var parentContextDirectory []byte
+	if req.ForkedFrom != nil {
+		// Can't use getExperimentAndCheckDoActions since model.Experiment doesn't have ParentArchived.
+		getTaskReq := &apiv1.GetTaskRequest{
+			TaskId: *req.ForkedFrom,
+		}
+		resp, err := a.GetTask(ctx, getTaskReq)
+		if err != nil {
+			return nil, err
+		}
+
+		parentConfig, err = resp.Task.Config.MarshalJSON()
+		if err != nil {
+			return nil, err
+		}
+
+		if len(req.ContextDirectory) == 0 {
```

Comment:
should it even be valid to have a context directory when you are forking?

Reply:
and if it is, is this the expected behavior? or would it be to merge them or something? maybe we just shouldn't allow it, for now. i don't think the cli supports it yet, right?

Reply:
This is done do that when the task is forked without providing a new context directory, we just get the one from the parent. We only use it for adding a new row to the task to context directory table.